packages/manager/modules/up-sell-cross-sell/src/advices/controller.js

```diff
@@ -0,0 +1,33 @@
+export default class AdvicesCtrl {
+  /* @ngInject */
+  constructor(atInternet) {
+    this.atInternet = atInternet;
+  }
+
+  $onInit() {
+    this.advicesAvailable = false;
+    this.loadAdvices();
+  }
+
+  loadAdvices() {
+    Promise.resolve(this.advices).then((advices) => {
```

Comment:
You should avoid using `Promise` while you're creating an AngularJS component

Reply:
Tt can take the promise that resolves to array or array directly. The advantage of passing a promise is it waits for it to resolve before rendering. it will not render advice if no devices are found. This simplifies the component usage like rowLoader in datagrid. Do you know a different way to do this?

Reply:
What I meant is to use `$q` instead of Javascrpit standard `Promise`. Sorry for confusion. 
No questionning the use of a promise here :D

Reply:
👍